go/tools/builders/nogo_main.go

```diff
@@ -542,6 +544,10 @@ func checkAnalysisResults(actions []*action, pkg *goPackage) ([]diagnosticEntry,
 		if baseConfig, ok := configs[nogoBaseConfigName]; ok {
 			currentConfig = baseConfig
 		}
+		if currentConfig.excludeFiles == nil {
+			// TODO(linzhp): add a test to verify the case of [].
+			currentConfig.excludeFiles = []*regexp.Regexp{_externalFilePattern}
```

Comment:
With this, most repos no longer need the JSON config unless they want to exclude certain paths in the repo. @fmeum